src/expr/src/explain/text.rs

```diff
@@ -298,15 +298,52 @@ impl MirRelationExpr {
                     })?;
                 }
             }
-            LetRec { ids, values, body } => {
-                let bindings = ids.iter().zip(values.iter()).collect::<Vec<_>>();
+            LetRec {
+                ids,
+                values,
+                max_iters,
+                body,
+            } => {
+                assert_eq!(ids.len(), values.len());
+                assert_eq!(ids.len(), max_iters.len());
+                let bindings = itertools::izip!(ids.iter(), values.iter(), max_iters.iter())
+                    .collect::<Vec<(&LocalId, &MirRelationExpr, &Option<u64>)>>(); // CLion needs these explicit types
```

Comment:
I'm not a huge fan of making our code specific to the limitations of specific IDEs!

I can't think of a good solution, but it makes my wonder why we are not running into this issue more often. Is this the only instance where CLion is not able to able to handle code that rustc is, or do you usually ignore errors? I assume more people are not using CLion, so it must happen from time to time that someone checks in code that's not compatible with it.

Reply:
Yes, there are some situations where it can't figure out some types. Here, I think the `izip` macro trips it up. There are also some situations where it detects a type error and makes a red underlining when in fact there is no type error. But these situations are rare enough that I can just ignore it with negligible impact on my productivity.

I deleted the explicit type annotation.